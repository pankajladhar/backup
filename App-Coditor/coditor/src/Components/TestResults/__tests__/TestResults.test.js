import React from "react";
import renderer from "react-test-renderer";
import TestResults from "./../TestResults";

describe("<TestResults /> ", () => {
  it("should render properly", () => {
    const tree = renderer.create(
      <TestResults
        results={
          '[{"scenario":"Should reverse the digits","status":"Fail"},{"scenario":"Should reverse the negative digits","status":"Pass"},{"scenario":"Should ignore the trailing zero","status":"Fail"}]'
        }
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render properly when results are not present", () => {
    const tree = renderer.create(<TestResults />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
