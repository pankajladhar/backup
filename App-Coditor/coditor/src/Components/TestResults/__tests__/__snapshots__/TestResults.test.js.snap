// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestResults />  should render properly 1`] = `
<div>
  <div
    className="mb-3"
  >
    <span
      className="font-bold text-green-500 mr-3"
    >
      1
       Passed
    </span>
    <span
      className="font-bold text-red-500"
    >
      2
       failed!
    </span>
  </div>
  <ul>
    <li
      className="p-2 flex items-center border-b border-gray-400"
    >
      <i
        className="icon-x-circle text-red-500 text-xl mr-2"
      />
      Should reverse the digits
    </li>
    <li
      className="p-2 flex items-center border-b border-gray-400"
    >
      <i
        className="icon-check-circle text-green-500 text-xl mr-2"
      />
      Should reverse the negative digits
    </li>
    <li
      className="p-2 flex items-center border-b border-gray-400"
    >
      <i
        className="icon-x-circle text-red-500 text-xl mr-2"
      />
      Should ignore the trailing zero
    </li>
  </ul>
</div>
`;

exports[`<TestResults />  should render properly when results are not present 1`] = `
<div>
  <div
    className="mb-3"
  >
    <span
      className="font-bold text-green-500 mr-3"
    >
      0
       Passed
    </span>
    <span
      className="font-bold text-red-500"
    >
      0
       failed!
    </span>
  </div>
  <ul />
</div>
`;
